Add Feat: 운동지수 정보 표시 기능 추가

wh5929 opened the pull request:
### 기능 추가
1. firestore 에서 weatherInfo data fetch 후 JustStartView에서 운동 지수 정보를 표시하는 기능 추가

### 추후 기능 추가 예정

1. 사용자의 위치 확인 후 위치에 따라 해당 지역의 운동지수 정보를 표시

### 구조 변경

1. 기존 : RunGuideViewModel 에서 weatherInfoRepository로 부터 데이터를 fetch -> 변경 후  : weatherInfoViewModel 생성 후 해당 뷰모델에서 데이터 처리를 담당

### 그 외 기타 소소한 코드 변경

wh5929 committed:
Check

wh5929 committed:
### 기능 추가

1. firestore 에서 weatherInfo data fetch 후 JustStartView에서 운동 지수 정보를 표시하는 기능 추가

### 추후 기능 추가 예정

1. 사용자의 위치 확인 후 위치에 따라 해당 지역의 운동지수 정보를 표시

### 구조 변경

1. 기존 : RunGuideViewModel 에서 weatherInfoRepository로 부터 데이터를 fetch -> 변경 후  : weatherInfoViewModel 생성 후 해당 뷰모델에서 데이터 처리를 담당

### 그 외 기타 소소한 코드 변경

hanjaesueng commented:
고생하셨습니다

DalioKim commented on GatheRunner/Presentaion/Running/Views/JustStartView/BottomButtonView.swift:
```diff
@@ -65,6 +65,12 @@ struct StartBtn: View {
     }
 }
 
+//struct WeatherInfoBtn: View {
```
[의견] 사용하시지 않는 코드의 주석은 제거하시는 편이 좋을 것 같습니다.

wh5929 replied:
확인 후 제거 했습니다.

DalioKim commented on GatheRunner/Presentaion/Running/ViewModel/RunGuideViewModel.swift:
```diff
@@ -5,54 +5,52 @@
 
-    // MARK: 서버 API 미구현 상태 -> 임시로 선언한 더미 데이터 -> API 구현 후 해당 코드 변경 예정
     private var runGuideExperienceItem = [
-        RunGuideItem(
```
[질문] 같은 코드인것같은데 삭제후 다시 추가하신 이유가 있으신가요?

wh5929 replied:
같은 코드이나 수정하다보니 이렇게 표시되었네요

DalioKim commented on GatheRunner/Presentaion/Running/Views/JustStartView/JustStartView.swift:
```diff
@@ -11,19 +11,46 @@ import SwiftUI
+extension JustStartView {
+    @ViewBuilder
+    private var weatherInfoDisplay: some View {
+        let weatherInfoString = isWeatherInfoDataNotNil ? ("오늘의 운동 지수 : " + getWeatherInfo.outdoorGrade! + (Int(getWeatherInfo.outdoorGrade!)! >= 70 ? " ☀️☀️ " : " ☁️☁️ ") + getWeatherInfo.content!) : ""
```
[의견] 옵셔널 강제해제는 바인딩방식으로 바꿔주시면 좋을 것 같습니다.

wh5929 replied:
변경했습니다.

DalioKim commented on GatheRunner/Presentaion/Running/Views/JustStartView/JustStartView.swift:
```diff
@@ -11,19 +11,46 @@ import SwiftUI
 
 struct JustStartView: View {
     @State private var isPresentedRunGuideDetailDescriptionView = false
+    @StateObject var getWeatherInfo: WeatherInfoViewModel = WeatherInfoViewModel()
```
[의견] 뷰모델자체의 네이밍은 적합한데, 뷰와 뷰모델의 네이밍이 좀더 연결되었으면 좋겟습니다.

wh5929 replied:
의견을 반영하여 수정했습니다.

wh5929 committed:
### PR Review 내용 반영

1. 클래스명 변경 WeatherInfoViewModel -> PrepareRunViewModel
2. 구조체명 변경 JustStartView -> PrepareRunView (운동 준비 뷰)
	2-1. View 타입 변수명 변경 -> weatherInfoDisplay -> workoutIndexView
	2-2. PrepareRunViewModel 이 다중호출 되는것을 막기 위해 PrepareRunView 의 onAppear Modifier 에서 PrepareRunViewModel의 getWorkoutIndex 함수(firestore에서 fetch한 데이터를 String으로 가공하여 받아오는 함수)를 호출하도록 함
	2-3. 매직넘버를 네임스페이스 상수를 사용하는것으로 변경
	2-4. PrepareRunView 내부에서 처리하던 텍스트처리를 PrepareRunViewModel 에서 처리하도록 변경

3. 불필요한 주석 코드 제거

wh5929 committed:
Merge remote-tracking branch 'origin/develop' into Refactor/2022102501
